src/planner/macos/mod.rs

```diff
@@ -96,29 +144,28 @@ impl Planner for Macos {
     }
 
     async fn plan(&self) -> Result<Vec<StatefulAction<Box<dyn Action>>>, PlannerError> {
+        if self.use_ec2_instance_store && !self.settings.determinate_nix {
+            return Err(PlannerError::Ec2InstanceStoreRequiresDeterminateNix);
+        }
+
         let root_disk = match &self.root_disk {
             root_disk @ Some(_) => root_disk.clone(),
             None => {
-                let buf = execute_command(
-                    Command::new("/usr/sbin/diskutil")
-                        .args(["info", "-plist", "/"])
-                        .stdin(std::process::Stdio::null()),
-                )
-                .await
-                .unwrap()
-                .stdout;
-                let the_plist: DiskUtilInfoOutput = plist::from_reader(Cursor::new(buf)).unwrap();
-
-                Some(the_plist.parent_whole_disk)
+                if self.use_ec2_instance_store {
+                    default_internal_root_disk().await?
+                } else {
+                    Some(default_root_disk().await?)
+                }
             },
         };
 
-        // The encrypt variable isn't used in the enterprise edition since we have our own plan step for it,
-        // however this match accounts for enterprise edition so the receipt indicates encrypt: true.
+        // The encrypt variable isn't used in Determinate Nix since we have our own plan step for it,
+        // however this match accounts for Determinate Nix so the receipt indicates encrypt: true.
         // This is a goofy thing to do, but it is in an attempt to make a more globally coherent plan / receipt.
-        let encrypt = match self.encrypt {
-            Some(choice) => choice,
-            None => {
+        let encrypt = match (self.settings.determinate_nix, self.encrypt) {
```

Comment:
Looks like we're carrying a bit of a diff here (git diff --no-ext-diff DetSys/main -- src/planner/macos/mod.rs) and I can't remember why. Not necessarily something we need to fix now but just flagging

Reply:
I think it's just evolving upstream?

When I blame the file on our main branch I only see Graham and Ana's names on the entire block, at least.

Reply:
Oh right guess I was comparing against a different commit than you merged